Feature/gocart drydep2

lholland-noaa opened the pull request.

lholland-noaa committed:
removing met state call

lholland-noaa committed:
adding dry dep to api

lholland-noaa committed:
adding dry dep to api

lholland-noaa committed:
added drydep to api dir's cmake file

lholland-noaa committed:
Adding cmake list for drydep back

lholland-noaa committed:
correcting more variable names in dry dep process

lholland-noaa committed:
improving consistency of dry dep variable names

lholland-noaa committed:
Adding drydep back to catchem lib

lholland-noaa committed:
modifications to organization, etc.

lholland-noaa committed:
implemented if statement for optional parameters

lholland-noaa committed:
REmoved arrows from merge

lholland-noaa committed:
resolving conflicts

lholland-noaa committed:
trying to resolve merge conflicts

lholland-noaa committed:
fixing inconsistency in variable name

lholland-noaa committed:
too many ifs

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update chemstate_mod.F90

Corrected variable name, tracer_index was duplicated

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Updated aeroindex to drydep index for chemstate variables

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Rearranging some of the if's and do statements to be more robust

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Re-arrange ordering in DryDepStateType

lholland-noaa committed:
added comment, marking where next step should be

bbakernoaa committed:
BB rework

bbakernoaa committed:
Adding rhop to function call

lholland-noaa committed:
Merge pull request #4 from bbakernoaa/feature/gocart_drydep2

Little help

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Removed extraneous end if

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Added resuspension flag back in.

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Adding comment to mark end of if statement

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Fixing typos

lholland-noaa committed:
Update Profile_AMAZON_Aug_06z.csv

added z0h

lholland-noaa committed:
Update Profile_AMAZON_Aug_12z.csv

added z0h

lholland-noaa committed:
Update Profile_AMAZON_Aug_18z.csv

Added z0h

lholland-noaa committed:
Update Profile_NCWCP.csv

added z0h

lholland-noaa committed:
Update Profile_NCWCP_Aug_06z.csv

added z0h

lholland-noaa committed:
Update Profile_NCWCP_Aug_12z.csv

added z0h

lholland-noaa committed:
Update Profile_NCWCP_Aug_18z.csv

added z0h

lholland-noaa committed:
Update Profile_SAHARA.csv

added z0h

lholland-noaa committed:
Update Profile_SAHARA_Aug_06z.csv

added z0h

lholland-noaa committed:
Update Profile_SAHARA_Aug_12z.csv

added z0h

lholland-noaa committed:
Update Profile_SAHARA_Aug_18z.csv

added z0h

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

cleaning up ccpr_scheme_gocart_drydep call

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

fixing typos

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Added missing commas

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Added missing commas

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Fixing order in subroutine call to CCPR_scheme_gocart_drydep

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Fixing call to ccpr_scheme_gocart_drydep for consistency

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

fixing call to ccpr_scheme_gocart_drydep

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

modifying call without resuspension

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Removing optional parameters to make them optional

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Removing duplicate radius, density; also made optional parameters optional

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Reordering call to ccpr_scheme_gocart_drydep

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Reordering call into subroutine

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

making optional variables optional?

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Removing unnecessary variables

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

updating intent for particle radius & density for resuspension case

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Revert back to original type for particle specs

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

filling out arguments in call to ccpr_scheme

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Updates to ccpr_scheme_gocart call

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

fixing typo in variable name

bbakernoaa commented on src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90:
```diff
@@ -65,21 +64,19 @@ subroutine CCPr_Scheme_GOCART_DryDep(km,
       REAL, pointer, intent(in), DIMENSION(:) :: hghte  ! Height [m]
-      REAL, pointer, intent(in)                      :: radius                                ! particle radius [m]
-      REAL, pointer, intent(in)                      :: rhop                                  ! particle density [kg/m^3]
+      REAL, pointer, intent(in), optional            :: radius                                ! particle radius [m]
```
I would move these back to required.  it isn't a problem just to pass all the variables to the scheme

lholland-noaa replied:
Done

bbakernoaa commented on src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90:
```diff
@@ -31,27 +31,26 @@ module CCPr_Scheme_GOCART_DryDep_Mod
+                                        drydepf,    &
+                                        rc,         &
+                                        ResuspensionOpt, &
+                                        radius,     &
```
see comment below

bbakernoaa commented on src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90:
```diff
@@ -65,21 +64,19 @@ subroutine CCPr_Scheme_GOCART_DryDep(km,
-      REAL, pointer, intent(in) :: gwettop                ! fraction soil moisture [1]
-      real, intent(in) :: radius 
-      real, intent(in) :: rhop
+      REAL, pointer, intent(in), optional :: u10m                   ! 10-m u-wind component [m/sec]
```
would make these required to the scheme itself

lholland-noaa replied:
done.

bbakernoaa commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -200,57 +200,59 @@ SUBROUTINE CCPr_DryDep_Run( MetState, DiagState, DryDepState, ChemState, RC )
-                                            RC, 
-                                            DryDepState%Resuspension,
-                                            )
+            call CCPr_Scheme_GOCART_DryDep( MetState%NLEVS,   &
```
I would suggest keeping this the way it was rather than inside the driver and making the other variables optional.  Let the scheme decide inside based on the passed DryDepState%Resuspension variable.

lholland-noaa replied:
done.

lholland-noaa commented:
That's fine, except it wasn't compiling with the fewer number of variables
passed to it in the no-resuspension case, and that was a failed attempt to
fix it.

On Thu, Sep 12, 2024 at 4:00 PM Barry Baker ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/process/drydep/CCPr_DryDep_Mod.F90
> <https://github.com/bbakernoaa/CATChem/pull/38#discussion_r1757503269>:
>
> > -                                            MetSTate%PBLH,
> -                                            MetState%HFLUX,
> -                                            VON_KARMAN,
> -                                            Cp,
> -                                            g0,
> -                                            radius,
> -                                            rhop
> -                                            MetState%U10M,
> -                                            MetSTate%V10M,
> -                                            MetState%FRACLAKE,
> -                                            MetState%GWETTO
> -                                            drydepf,
> -                                            RC,
> -                                            DryDepState%Resuspension,
> -                                            )
> +            call CCPr_Scheme_GOCART_DryDep( MetState%NLEVS,   &
>
> I would suggest keeping this the way it was rather than inside the driver
> and making the other variables optional. Let the scheme decide inside based
> on the passed DryDepState%Resuspension variable.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bbakernoaa/CATChem/pull/38#pullrequestreview-2301295161>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJUZGYYCA7SWGQC4IB2DZWHXEBAVCNFSM6AAAAABOBKGGPKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMBRGI4TKMJWGE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

zmoon committed:
Merge branch 'begins' into drydep

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Got rid of divergent calls to ccpr_scheme_gocart_drydep

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Getting rid of "optional" variables

lholland-noaa committed:
Update CMakeLists.txt

testing GOCART2G vs. ${_lib}

lholland-noaa committed:
Update CMakeLists.txt

Revert back

lholland-noaa committed:
Update CMakeLists.txt

removed gocart from target link

lholland-noaa committed:
Update CMakeLists.txt

Modifying linked library list

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

fixing typos

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Fixed typos, removed unnecessary variables

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

fixed more typos

bbakernoaa committed:
getting drydep to compile

lholland-noaa committed:
Merge branch 'feature/gocart_drydep2' into feature/drydep_compile

lholland-noaa committed:
Merge pull request #5 from bbakernoaa/feature/drydep_compile

getting drydep to compile

zmoon committed:
Really LWI is integer

zmoon committed:
Not used (just state mod)

zmoon committed:
Test almost compiling

zmoon committed:
Builds now

zmoon committed:
Allocate

lholland-noaa committed:
Update CATChem_config.yml

Adding drydep activate

lholland-noaa committed:
Update CATChem_config.yml

adding resuspension: false

lholland-noaa committed:
Update config_opt_mod.F90

changing "DryDep" to "drydep" (all lowercase) - getting rid of camelcase.

lholland-noaa committed:
Update config_mod.F90

getting rid of camel case in "DryDep"

lholland-noaa committed:
Update CATChem_species.yml

Adding is_drydep to dust species

lholland-noaa committed:
Update test_drydep.f90

Updates to file locations, variable settings, etc.

lholland-noaa committed:
Update test_drydep.f90

Update resuspension test to more reasonable numbers

lholland-noaa committed:
Update test_drydep.f90

Commenting out loop and changing everything to lowercase

lholland-noaa committed:
Update test_drydep.f90

Testing drydep frequency change (not an array)

lholland-noaa committed:
Update test_drydep.f90

Reverting back, but changing index

lholland-noaa committed:
Update test_drydep.f90

reverting indexing

lholland-noaa committed:
Update test_drydep.f90

Making capitalization more consistent

lholland-noaa committed:
Update test_drydep.f90

Updating default values for test

lholland-noaa committed:
Update test_drydep.f90

More edits

lholland-noaa committed:
Update diagstate_mod.F90

Updating dry dep portion to include nSpeciesAeroDryDep instead of nSpeciesAero

lholland-noaa committed:
Update metstate_mod.F90

Initializing thermal roughness length to zero

lholland-noaa committed:
Update test_drydep.f90

Correcting LWI to land type (integer 1)

lholland-noaa committed:
Update test_drydep.f90

Updating MAIRDEN to more reasonable value.

lholland-noaa committed:
Update test_drydep.f90

Made a mistake and reverting

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

changing airden to mairden (moist air density)

lholland-noaa committed:
Update test_drydep.f90

corrected scheme option setting

lholland-noaa committed:
Update metstate_mod.F90

Changing LWI to real (instead of integer, despite it describing a trinary flag)

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

changing LWI to type real

lholland-noaa committed:
resolving conflict

lholland-noaa committed:
Fixing how DryDepState SchemeOpt is set

lholland-noaa committed:
Update test_drydep.f90

Removing extra allocate(MetState$MAIRDEN(...))

lholland-noaa committed:
Update test_drydep.f90

Reverting to commenting out the print_info

lholland-noaa committed:
adding dry dep aerosols into dry dep summary

lholland-noaa committed:
adding default for drydep scheme

lholland-noaa committed:
fixing typos

lholland-noaa committed:
attempting to use reshape to assign to a pointer

lholland-noaa committed:
rethinking value assignment method

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

removing reshape

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

testing if pointer is allocated

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

testing pointer assignment

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Testing new if statement for size(shape()) of temperature before assigning to gocart_tmpu pointer

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Removing unnecessary if statements

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

adding target for pointer

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

testing reshape of target variable for pointer

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

testing additional variables

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

left out hghte from pointer values

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

fixing syntax for reshape

lholland-noaa committed:
Update test_drydep.f90

temporarily commenting out the call to assert, testing the drydep_frequency

lholland-noaa committed:
Update test_drydep.f90

temporarily turning off print_info and assert for test3

lholland-noaa committed:
Update test_drydep.f90

More reasonable numbers for resuspension test

lholland-noaa committed:
restoring print_info calls (from comments)

lholland-noaa committed:
changing from moist air density to dry air density, upon closer inspection of dry dep external subroutine

lholland-noaa committed:
fixed a typo in config file, making capitalization more consistent

lholland-noaa committed:
modified print_info in test_drydep

lholland-noaa committed:
updating default drydepscheme opt values

lholland-noaa committed:
changes to config file read-in test

lholland-noaa committed:
edits to print_info in cases of unassigned variables

lholland-noaa committed:
writing out completion of tests

lholland-noaa committed:
adding more detail prior to calls to print_info

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Allocating gocart_tmpu and other null pointers before assigning values

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Allocating allocatable arrays before assigning values.

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Removing previous statements

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Using reshape on GOCART_LWI

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

testing GOCART_LWI assignment

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Testing same for GOCART_USTAR

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Updating all gocart variables to be pointers

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Fixing oopsy in rank for U10, V10, etc.

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Additional comments

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

specifying pointer targets

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

deallocating optional parameters before call

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

using "free" instead

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update test_drydep.f90

removing superfluous commas in write statement

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Changing Zmid to Z

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update test_drydep.f90

ordering nlevs top down

lholland-noaa committed:
Update test_drydep.f90

Adding test values for wind speed

lholland-noaa committed:
Update test_drydep.f90

Adding additional test data temporarily

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Fixing comment

lholland-noaa committed:
Update test_drydep.f90

Edits to values

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

fixing syntax in deallocate statement

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Improving logic for optional variables.  Still need to fix where particle radius and density are set.

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update test_drydep.f90

temporarily comment out call to version with resuspension turned off until handling of pointers for optional variables is sorted out

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update test_drydep.f90

lholland-noaa committed:
Update test_drydep.f90

lholland-noaa committed:
Update test_drydep.f90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

temporarily adding debugging statements

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update test_drydep.f90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

bbakernoaa commented:
@lholland-noaa is this ready to go?

lholland-noaa commented:
Yes, it is.

On Sun, Sep 29, 2024 at 6:07 PM Barry Baker ***@***.***>
wrote:

> @lholland-noaa <https://github.com/lholland-noaa> is this ready to go?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bbakernoaa/CATChem/pull/38#issuecomment-2381627747>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJU7GWZP2RDJBF2HNNXLZZB2YDAVCNFSM6AAAAABOBKGGPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGOBRGYZDONZUG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

bbakernoaa committed:
pre-commit

bbakernoaa commented:
@zmoon please take a Quick Look.  I did the pre-commit formatting and all tests are passing.

bbakernoaa reviewed (approved):
@lholland-noaa Great work!

lholland-noaa commented:
Thank you so much!

On Mon, Sep 30, 2024 at 9:43 AM Barry Baker ***@***.***>
wrote:

> ***@***.**** approved this pull request.
>
> @lholland-noaa <https://github.com/lholland-noaa> Great work!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bbakernoaa/CATChem/pull/38#pullrequestreview-2337605497>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJU2RU3HQRDPNJ6I3ZPTZZFIQFAVCNFSM6AAAAABOBKGGPKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGMZXGYYDKNBZG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

zmoon reviewed (commented):
Some warnings to address, to keep our warnings down:
* unused declared variables in `Config_Process_DryDep`
* some unused variables in GOCART DryDep mod
* some unused arguments in DryDep mod Init/Run/Finalize

DryDep test has no assertions about the results. We've been trying to add at least one for each case. For example for dust:
https://github.com/bbakernoaa/CATChem/blob/dd1dc7174e67f6550395a9b7c4b95a7c04f11343/tests/test_dust.f90#L160

lholland-noaa committed:
Update test_drydep.f90

Uncommented print_info and assert comments

lholland-noaa committed:
Update config_mod.F90

Removing unused variables: a_int, a_str, C, N, nsubstrs, substrs, v_real, v_str, etc.

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Removing unused variable, GOCART_DRYDEPF

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Removed/commented out unused variables

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Removed unused variables, uncommented references to diagstate.

lholland-noaa committed:
Update diagstate_mod.F90

Updating indices of DiagState%drydep_frequency and DiagState%drydep_vel

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Adding correct indices for drydepf

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Commenting out Deallocation for now (as nothing was allocated during the init step??)

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

debugging the diagstate

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

more debugging

lholland-noaa committed:
Update diagstate_mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

more debugging

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update diagstate_mod.F90

more debugging

lholland-noaa committed:
Update diagstate_mod.F90

more debugging

lholland-noaa committed:
Update diagstate_mod.F90

debugging

lholland-noaa committed:
Update diagstate_mod.F90

continuing to debug

lholland-noaa committed:
Update test_drydep.f90

More debugging

lholland-noaa committed:
Update test_drydep.f90

lholland-noaa committed:
Update test_drydep.f90

lholland-noaa committed:
Update test_drydep.f90

RAE is missing, removed references to it.

lholland-noaa committed:
Update test_drydep.f90

lholland-noaa committed:
Update diagstate_mod.F90

commenting out debugging comments

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

adding drydepstate for freq and vel

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Adding drydep freq to DryDepState type as allocatable array

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

bbakernoaa commented on tests/test_drydep.f90:
```diff
@@ -0,0 +1,178 @@
+      call cc_emit_error(errMsg, rc, thisLoc)
+      stop 1
+   end if
+
```
```suggestion
! Allocate DiagState
   call cc_allocate_diagstate(Config, DiagState, ChemState, RC)
   if (rc /= CC_SUCCESS) then 
      errMsg = 'Error in cc_allocate_diagstate'
      stop 1
   endif
```

bbakernoaa commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,304 @@
+                  !print *, "drydepf(1,1) = ", drydepf(1,1)
+                  DryDepState%drydep_frequency(ChemState%DryDepIndex(i)) = drydepf(1,1)
+                  DryDepState%drydep_vel(ChemState%DryDepIndex(i)) = MetState%ZMID(1) * drydepf(1,1)
+                  !DiagState%drydep_frequency= DryDepState%drydep_frequency(ChemState%DryDepIndex(i))
```
These two lines need uncommented as well 

```suggestion
                  DiagState%drydep_frequency(i) = drydepf(1,1)
                  DiagState%drydep_vel(i) = MetState%ZMID(1) * drydepf(1,1)
```

bbakernoaa committed:
allocate diagstate

bbakernoaa committed:
remove diagnostic printing

bbakernoaa committed:
add error checking to allocate status and deallocate

bbakernoaa commented:
@zmoon I think everything should be ready.  Give it another look please

bbakernoaa commented:
@lholland-noaa Great work addressing the Zach's comments

zmoon commented on src/core/metstate_mod.F90:
```diff
@@ -61,7 +61,7 @@ MODULE MetState_Mod
       !---------------------
       REAL(fp)              :: AREA_M2         !< Grid box surface area [m2]
-      INTEGER               :: LWI             !< Land water ice mask (0-sea, 1-land, 2-ice)
+      REAL(fp)              :: LWI             !< Land water ice mask (0-sea, 1-land, 2-ice)
```
This is really integer and shouldn't be changed here. You can convert it before passing or as passing to GOCART as necessary later, like `real(lwi, kind=fp)`.

zmoon replied:
For example, `CCPr_Scheme_GOCART_DryDep` could take the integer LWI as input, but in the prep
```f90
GOCART_LWI = real(LWI, kind=fp)
```

lholland-noaa replied:
In principle, I agree with you.  It is a flag (!!), yet all of GOCART's functions expect a real.  I'm not sure if this trinary flag will be used anywhere but GOCART.

lholland-noaa replied:
REsolved - took the advice to change it (back?) to integer, then retype as real later.

zmoon commented on src/CMakeLists.txt:
Not that it really matters, but the previous order makes more sense to me.

zmoon commented on src/api/catchem.F90:
```diff
@@ -77,6 +77,11 @@ module CATChem
    use CCPr_Plumerise_mod, only: cc_plumerise_run => CCPr_Plumerise_Run           !< Plumerise Process Run Routine
    use CCPr_Plumerise_mod, only: cc_plumerise_finalize => CCPr_Plumerise_Finalize !< Plumerise Process Finalization Routine
+   ! Dry Dep
+   use CCPr_DryDep_mod, only: DryDepStateType                               !< DryDep State
```
```suggestion
   use CCPr_DryDep_mod, only: DryDepStateType                            !< DryDep State
```

zmoon commented on src/core/chemstate_mod.F90:
```diff
@@ -100,11 +101,13 @@ subroutine Chem_Allocate(GridState, ChemState, RC)
       ! USES
       USE GridState_Mod,  ONLY : GridStateType
       USE Species_Mod,    Only : SpeciesType
+      USE Metstate_Mod
```
```suggestion
```
seems not needed

zmoon commented on src/core/config_mod.F90:
```diff
@@ -1276,4 +1287,90 @@ SUBROUTINE Config_Process_Plumerise( ConfigInput, Config, RC )
+      thisLoc = ' -> at Config_Process_DryDep (in CATChem/src/core/config_mod.F90)'
+      errMsg = ''
+
+      ! TODO #105 Fix reading of config file
```
`#105` ?

lholland-noaa replied:
No idea where that came from ... @bbakernoaa  ???

bbakernoaa replied:
Maybe something I did by accident.  I would just remove the comment

lholland-noaa replied:
Removed.

zmoon commented on src/core/config_opt_mod.F90:
```diff
@@ -94,7 +97,10 @@ MODULE Config_Opt_Mod
+      ! DryDeposition Process
+      LOGICAL                     :: drydep_activate
+      INTEGER                     :: drydep_scheme
+      LOGICAL                     :: drydep_resuspension  ! Turn on resuspension
```
```suggestion
      LOGICAL                     :: drydep_resuspension  !< Turn on resuspension
```

zmoon commented on src/core/diagstate_mod.F90:
```diff
@@ -39,6 +47,10 @@ module DiagState_Mod
       real(fp) :: sofiev_plumerise_height !< Effective plume rise height from Sofiev algorithm [m]
 
+
+      real(fp), allocatable :: drydep_frequency(:)
+      real(fp), allocatable :: drydep_vel(:)
```
Suggest either `drydep_freq` and `drydep_vel` OR `drydep_frequency` and `drydep_velocity`

zmoon replied:
Also considering documenting them with `!< end of line comment`s, as this helps with the IDE experience.

zmoon commented on src/core/diagstate_mod.F90:
```diff
@@ -91,6 +105,27 @@ subroutine Diag_Allocate(Config, DiagState, RC)
+
+         Allocate(DiagState%drydep_vel(ChemState%nSpeciesAeroDryDep), STAT=RC)
+         IF ( RC /= CC_SUCCESS ) THEN
+            ErrMsg = 'Could not Allocate ChemState%ChemSpecies(i)%conc'
```
```suggestion
            ErrMsg = 'Could not Allocate DiagState%drydep_vel(ChemState%nSpeciesAeroDryDep)'
```

zmoon commented on src/core/gridstate_mod.F90:
```diff
@@ -23,6 +23,7 @@ module GridState_Mod
       integer :: ny = 1
       integer :: number_of_levels  !< The number of vertical levels
       integer :: number_of_soil_layers  !< The number of soil layers
+      integer :: model_timestep
```
This doesn't seem to be used anywhere.
```suggestion
```

lholland-noaa commented:
Thanks for taking it across the finish line!

On Wed, Oct 2, 2024 at 11:53 AM Barry Baker ***@***.***>
wrote:

> @lholland-noaa <https://github.com/lholland-noaa> Great work addressing
> the Zach's comments
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bbakernoaa/CATChem/pull/38#issuecomment-2389028211>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJU6BDUCXP44HSXRKHMTZZQJF5AVCNFSM6AAAAABOBKGGPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGOBZGAZDQMRRGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+!! \author Lacey Holland
+!! \date 07/2024
+!!!>
+MODULE CCPR_DryDep_mod
```
We should probably update the codebase for consistency at some point, for `CCPR` vs `CCPr`. `CCPr` seems more common, but several cases of `CCPR` as well.

zmoon replied:
Don't need to do it here though.

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+      real                            :: particleradius
+      real                            :: particledensity
+      real, allocatable               :: drydep_frequency(:)      ! could have one per chem species, revisit later
+      real, allocatable               :: drydep_vel(:)            ! could have one per chem species, revisit later
```
Consider documenting with `!<` comments all of these, including units

lholland-noaa replied:
I added documentation for particle radius, density

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+            ErrMsg = 'Could not Allocate DryDepState%drydep_frequency(ChemState%nSpeciesAeroDryDep)'
+            CALL CC_Error( ErrMsg, RC, ThisLoc )
+         ENDIF
+         DryDepState%drydep_frequency(1:ChemState%nSpeciesAeroDryDep)=0
```
Could use the special `ZERO` here?

lholland-noaa replied:
changed.

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+            ErrMsg = 'Could not Allocate DryDepState%drydep_vel(ChemState%nSpeciesAeroDryDep)'
+            CALL CC_Error( ErrMsg, RC, ThisLoc )
+         ENDIF
+         DryDepState%drydep_vel(1:ChemState%nSpeciesAeroDryDep)=0
```
Could use the special `ZERO` here?`

lholland-noaa replied:
changed

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+         if (Config%drydep_scheme==1) then
+            DryDepState%SchemeOpt = 1
+         else
+            DryDepState%SchemeOpt = 1
```
Maybe print a warning so it's clear that the input setting isn't being respected

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+            DryDepState%Resuspension = .true.
+         else
+            DryDepState%Resuspension = .false.
+         end if
```
This should do the same thing
```suggestion
         DryDepState%Resuspension = Config%drydep_resuspension
```

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+         !------------------
+         ! For now, the only option is SchemeOpt = 1
+         ! Yes, this is hard-coded for now
+
```
```suggestion
```

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+      ! USE
+      USE ChemState_Mod, ONLY : ChemStateType
+      use Config_Opt_Mod, only : ConfigType
+      use DiagState_Mod, only : DiagStateType
```
Already `use`d at module level
```suggestion
```

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+      USE precision_MOD
+      use MetState_Mod, only : MetStateType
+      use DiagState_Mod, only : DiagStateType
+      use ChemState_Mod, only : ChemStateType
```
Already `use`d at module level
```suggestion
```

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+   SUBROUTINE CCPr_DryDep_Run( MetState, DiagState, DryDepState, ChemState, RC )
+
+      ! USE
+      USE constants
```
```suggestion
      USE constants, only : Cp, g0, VON_KARMAN
```
I think these are the only ones used, but please confirm

zmoon replied:
Also, some of these might be better off just imported in `CCPr_Scheme_GOCART_DryDep` instead?

* von Karman probably doesn't need to be an input
* $c_{pd}$ technically does vary with temperature and pressure, so it may make sense to keep it as an input. But the variations are small for typical lowest model layer conditions.
* it seems like the scheme actually wants $g$, not $g_0$. Technically $g$, unlike $g_0$ varies with height above sea level, e.g. but since it's only for the lowest layer, it probably doesn't need to be an input

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+         if (DryDepState%SchemeOpt == 1) then
+            ! Run the DryDep Scheme - Only Applicable to AEROSOL species
+            !-------------------------
+            if (ChemState%nSpeciesAeroDryDep > 0) then
```
This guard could be removed. No looping will occur if `ChemState%nSpeciesAeroDryDep` isn't $\ge 1$

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+                  !!!!FIXME: COME BACK TO THIS LATER
+                  !print *, "size(DiagState%drydep_frequency)=", size(DiagState%drydep_frequency)
+                  !print *, "ChemState%DryDepIndex(i)=", ChemState%DryDepIndex(i)
+                  !print *, "drydepf(1,1) = ", drydepf(1,1)
```
Is it safe to remove this now?
```suggestion
```

zmoon commented on src/process/drydep/CCPr_DryDep_Mod.F90:
```diff
@@ -0,0 +1,320 @@
+                  DiagState%drydep_vel(i) = MetState%ZMID(1) * drydepf(1,1)
+
+                  ! apply drydep velocities/freq to chem species
+                  dqa = 0.
```
```suggestion
```
next-next line ensures `dqa` is set to $\ge 0$

zmoon commented on src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90:
```diff
@@ -0,0 +1,272 @@
+
+      ! transform data for GOCART DryDeposition call
+
+      call PrepMetVarsForGOCART(km,     &
```
I don't think this needs to be a separate subroutine. The prep (allocating the pointers and then assigning to them) can be done in the body of the subroutine.

zmoon commented on src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90:
```diff
@@ -0,0 +1,272 @@
+      z0h,             &
+      drydepf,         &
+      ResuspensionOpt, &
+      radius,          &
+      rhop,            &
```
Suggest moving these optional args to the end (or making them not optional)

zmoon commented on src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90:
```diff
@@ -0,0 +1,272 @@
+      integer, intent(out) :: RC                      ! Success or Failure
+
+      ! Local Variables
+      real, pointer :: GOCART_tmpu(:,:,:)
+      real, pointer :: GOCART_rhoa(:,:,:)
+      real, pointer :: GOCART_HGHTE(:,:,:)
+      real, pointer :: GOCART_LWI(:,:)
+      real, pointer :: GOCART_USTAR(:,:)
+      real, pointer :: GOCART_PBLH(:,:)
+      real, pointer :: GOCART_HFLUX(:,:)
+      real, pointer :: GOCART_Z0H(:,:)
+      real, pointer :: GOCART_U10(:,:)
+      real, pointer :: GOCART_V10(:,:)
+      real, pointer :: GOCART_FRACLAKE(:,:)
+      real, pointer :: GOCART_GWETTOP(:,:)
```
Suggest nullifying all of these in declaration (add ` => null()`). Then the later `nullify()` is not needed.

zmoon replied:
Also probably best to explicitly de-allocate them at the end of the subroutine. Or I read that if you nullify them at the end, the anonymous target memory will also be automatically de-allocated. This may be better since it will also work for the ones that haven't been allocated (e.g. skipped for non-resuspension run).

lholland-noaa replied:
should be resolved now.

zmoon commented on src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90:
```diff
@@ -0,0 +1,272 @@
+      allocate(GOCART_HFLUX(1, 1))
+      allocate(GOCART_Z0H(1, 1))
+
+      GOCART_TMPU = reshape(tmpu, (/1, 1, km/)) ! temperature [K]
```
I think this way should avoid an allocation for the reshape result
```suggestion
      GOCART_TMPU(1,1,:) = tmpu ! temperature [K]
```
(similarly for the other two)

zmoon commented on tests/MetProfiles/Profile_AMAZON_Aug_06z.csv:
```diff
@@ -90,6 +90,8 @@ wilt,1
 0.0000e+00
 z0,1
 3.5208e-05
+z0h,1
+3.5208e-05
```
@bbakernoaa how did you add this? I see it's just equal to z0. Does the sfc output have z0h?

Either way, `vars.yml` should be modified and the testing mod modified (should happen automatically when running `get-vars.py`) to automatically load it into MetState.

bbakernoaa replied:
I don't think that sfc output has z0h.  most of the time z0h could be equivalent to z0 as a rough estimate

zmoon commented on tests/test_drydep.f90:
```diff
@@ -0,0 +1,185 @@
+   end if
+
+   call print_info(Config, DryDepState, MetState, title)
+   call assert(DiagState%drydep_frequency(1) > 0.0_fp, "Test 2 GOCART drydep Scheme (resuspension activated)")
```
The first result ("Test 2") is usually the same as the second one ("Test 3"):
```
6:  DryDepState%drydepf =    4.66378151E-06   4.66378151E-06
```
but _sometimes_, I'm getting this for the second one:
```
6:  DryDepState%drydepf =    9.99999997E-07   4.66378151E-06
```
which indicates an issue somewhere (see other comment lower in this file about GWETTOP and FRLAKE).

zmoon replied:
In any case, _should_ we expect the result to be different with resuspension activated? If so, then either a fix or different input parameters are needed (e.g. faster winds).

lholland-noaa replied:
I think this needs revisiting.  The 2nd one should only have one value returned.  Maybe this is another case of deallocating/nullifying needs to be done at the beginning and end of the subroutine.

zmoon replied:
Yeah I think that 2nd value is just there from the first run, same DryDepState being re-used.

lholland-noaa replied:
I think this has been resolved.  If not, let's revisit next time.  I'll add a comment to revisit it for now!

zmoon commented on tests/test_drydep.f90:
```diff
@@ -0,0 +1,185 @@
+   !write (*,*), '-- '
+   !write (*,*), 'Completed ', title
+   !write (*,*), '--'
+
```
```suggestion
   call print_info(Config, DryDepState, MetState, title)
   call assert(DiagState%drydep_frequency(1) > 0.0_fp, "Test GOCART DryDep Scheme (no resuspension)")

```

zmoon commented on tests/test_drydep.f90:
```diff
@@ -0,0 +1,185 @@
+         write(*,*) 'DryDepState%SchemeOpt = ', DryDepState_%SchemeOpt
+
+         if (DryDepState_%Resuspension) then
+            write(*,*) 'MetState%GWETTOP =', MetState_%GWETTOP
```
Note GWETTOP isn't set above.

zmoon replied:
Upon setting it to something (I used 0.2), the second `DiagState%drydep_frequency(1)` result becomes random.

zmoon replied:
The solution seems to be that setting FRLAKE is also needed (and can be added to the print here as well). For some reason, on my machine GWETTOP is always auto initialized to the same number (1.40129846E-45), but FRLAKE gets auto initialized to something random.

lholland-noaa replied:
Added.

zmoon commented on tests/test_drydep.f90:
```diff
@@ -0,0 +1,185 @@
+      if (DryDepState%Activate) then
+
+         write(*,*) 'DryDepState%Activate = ', DryDepState_%Activate
+         write(*,*) 'DryDepState%SchemeOpt = ', DryDepState_%SchemeOpt
```
Suggest adding `DryDepState_%Resuspension` as well

lholland-noaa replied:
I'm sorry - I think I resolved some of these too soon and without address the change!  I clicked too fast!!

zmoon replied:
Feel free to unresolve if you find one

lholland-noaa replied:
Added.

zmoon commented on tests/test_drydep.f90:
```diff
@@ -0,0 +1,185 @@
+      write(*,*) 'Config%drydep_scheme = ', Config_%drydep_scheme
+      write(*,*) 'Config%drydep_resuspension = ', Config_%drydep_resuspension
+
+      if (DryDepState%Activate) then
```
```suggestion
      if (DryDepState_%Activate) then
```
doesn't make a difference, but consistency

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

Added comments on units/explanation of variables

lholland-noaa committed:
Update CCPr_DryDep_Mod.F90

change 0 to ZERO

lholland-noaa committed:
Update tests/test_drydep.f90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update tests/test_drydep.f90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/core/chemstate_mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

added deallocation of GOCART pointers

lholland-noaa commented:
Done.

On Wed, Oct 2, 2024 at 5:24 PM Zachary Moon ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90
> <https://github.com/bbakernoaa/CATChem/pull/38#discussion_r1785265248>:
>
> > +      real, pointer :: GOCART_tmpu(:,:,:)
> +      real, pointer :: GOCART_rhoa(:,:,:)
> +      real, pointer :: GOCART_HGHTE(:,:,:)
> +      real, pointer :: GOCART_LWI(:,:)
> +      real, pointer :: GOCART_USTAR(:,:)
> +      real, pointer :: GOCART_PBLH(:,:)
> +      real, pointer :: GOCART_HFLUX(:,:)
> +      real, pointer :: GOCART_Z0H(:,:)
> +      real, pointer :: GOCART_U10(:,:)
> +      real, pointer :: GOCART_V10(:,:)
> +      real, pointer :: GOCART_FRACLAKE(:,:)
> +      real, pointer :: GOCART_GWETTOP(:,:)
>
> Also probably best to explicitly de-allocate them at the end of the
> subroutine.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bbakernoaa/CATChem/pull/38#discussion_r1785265248>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJU3C3WWRZUFDDWZPETTZZRP7JAVCNFSM6AAAAABOBKGGPKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGNBUGEZTANZQG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

lholland-noaa commented:
It took a couple of tries, but done now.

On Thu, Oct 3, 2024 at 8:50 AM Lacey Holland - NOAA Federal <
***@***.***> wrote:

> Done.
>
> On Wed, Oct 2, 2024 at 5:24 PM Zachary Moon ***@***.***>
> wrote:
>
>> ***@***.**** commented on this pull request.
>> ------------------------------
>>
>> In src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90
>> <https://github.com/bbakernoaa/CATChem/pull/38#discussion_r1785265248>:
>>
>> > +      real, pointer :: GOCART_tmpu(:,:,:)
>> +      real, pointer :: GOCART_rhoa(:,:,:)
>> +      real, pointer :: GOCART_HGHTE(:,:,:)
>> +      real, pointer :: GOCART_LWI(:,:)
>> +      real, pointer :: GOCART_USTAR(:,:)
>> +      real, pointer :: GOCART_PBLH(:,:)
>> +      real, pointer :: GOCART_HFLUX(:,:)
>> +      real, pointer :: GOCART_Z0H(:,:)
>> +      real, pointer :: GOCART_U10(:,:)
>> +      real, pointer :: GOCART_V10(:,:)
>> +      real, pointer :: GOCART_FRACLAKE(:,:)
>> +      real, pointer :: GOCART_GWETTOP(:,:)
>>
>> Also probably best to explicitly de-allocate them at the end of the
>> subroutine.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/bbakernoaa/CATChem/pull/38#discussion_r1785265248>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BHNBJU3C3WWRZUFDDWZPETTZZRP7JAVCNFSM6AAAAABOBKGGPKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGNBUGEZTANZQG4>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
>

lholland-noaa committed:
Update src/api/catchem.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update test_drydep.f90

Added write out of DryDepState%Resuspension

lholland-noaa committed:
Update test_drydep.f90

Added MetState%GWETTOP, MetState%FRLAKE

lholland-noaa committed:
Update src/core/config_opt_mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/core/diagstate_mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/process/drydep/CCPr_DryDep_Mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/core/gridstate_mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/process/drydep/CCPr_DryDep_Mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/process/drydep/CCPr_DryDep_Mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/process/drydep/CCPr_DryDep_Mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/process/drydep/CCPr_DryDep_Mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/process/drydep/ccpr_scheme_GOCART_DryDep_mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa committed:
Update src/process/drydep/CCPr_DryDep_Mod.F90

Co-authored-by: Zachary Moon <zmoon92@gmail.com>

lholland-noaa commented:
I think I copied the exact value of z0 into z0h


On Mon, Oct 7, 2024 at 3:14 PM Barry Baker ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tests/MetProfiles/Profile_AMAZON_Aug_06z.csv
> <https://github.com/bbakernoaa/CATChem/pull/38#discussion_r1790747367>:
>
> > +z0h,1
> +3.5208e-05
>
> I don't think that sfc output has z0h. most of the time z0h could be
> equivalent to z0 as a rough estimate
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/bbakernoaa/CATChem/pull/38#discussion_r1790747367>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHNBJU7TUWTNYYINP7BCBKTZ2LMQDAVCNFSM6AAAAABOBKGGPKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDGNJSG42DCNJWGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Lacey Holland, Ph.D.
Research Physical Scientist
Atmospheric Sciences Modeling Division (ASMD)
NOAA/OAR/ARL
NCWCP, Rm 4257
5830 University Research Court
College Park, Maryland  20740
‪(301) 427-7880‬

lholland-noaa committed:
Update metstate_mod.F90

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Updating LWI to Integer, then GOCART_LWI is real

lholland-noaa committed:
Update config_mod.F90

Removed comment #105 :-)

lholland-noaa committed:
Update test_drydep.f90

Added comment to revisit

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Removing "optional"

lholland-noaa committed:
Update ccpr_scheme_GOCART_DryDep_mod.F90

Fixing typecast to real for LWI
